src/Illuminate/Console/Command.php

```diff
@@ -186,10 +193,85 @@ public function run(InputInterface $input, OutputInterface $output)
      * @param  \Symfony\Component\Console\Input\InputInterface  $input
      * @param  \Symfony\Component\Console\Output\OutputInterface  $output
      * @return mixed
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        return $this->laravel->call([$this, 'handle']);
+        if ($this->hasValidInput()) {
+            return $this->laravel->call([$this, 'handle']);
+        }
+
+        return $this->displayValidationErrors();
+    }
+
+    /**
+     * Check if the argument and option values are valid.
+     *
+     * @return bool
+     *
+     * @throws \Illuminate\Contracts\Container\BindingResolutionException
+     */
+    protected function hasValidInput()
+    {
+        if (empty($rules = $this->rules())) {
+            return true;
+        }
+
+        $this->validator = $this->validator ?? $this->laravel->make('validator')->make(
```

Comment:
Reference the interface here instead of the string alias. Maybe move the resolving to a separate method.

Reply:
Referencing interface now, and moved make validator to its own method.